File: scripts/GetVariableList.java
/***************** GetVariableList.java ******************/
package Lupremica;

import java.util.List;
import net.sourceforge.waters.model.base.Proxy;
import net.sourceforge.waters.subject.module.ModuleSubject;
import net.sourceforge.waters.model.module.VariableComponentProxy;
import org.apache.logging.log4j.Logger;

public class GetVariableList
{
	public GetVariableList(org.supremica.gui.ide.IDE ide) // called by RunScript
	throws ClassNotFoundException
	{
		final Logger logger = ide.getTheLog();
		logger.info("Get variable list of current module 0");

		final ModuleSubject module = ide.getActiveDocumentContainer().getEditorPanel().getModuleSubject();
		List<VariableComponentProxy> variableList = GetVariableList.getVariableComponentProxys(module);
		for(final VariableComponentProxy var : variableList)
		{
			logger.info(var.getInitialStatePredicate() + " (" + var.getType() + ")");
		}
	}
	public static List<VariableComponentProxy> getVariableComponentProxys(final ModuleSubject module)
	{
		final List<VariableComponentProxy> variableList = new java.util.ArrayList<>();
		for(final Proxy proxy : module.getComponentList())
		{
			// All four of these if-clauses work as expected, but none work in Lua...
			// if(proxy.getProxyInterface().getSimpleName().equals("VariableComponentProxy"))
			// if(proxy instanceof VariableComponentProxy)
			// if(Class.forName("net.sourceforge.waters.model.module.VariableComponentProxy").isInstance(proxy))
			if(VariableComponentProxy.class.isInstance(proxy))
			{
				final VariableComponentProxy var = (VariableComponentProxy)proxy;
				variableList.add(var);
			}
		}
		return variableList;
	}
}

File: scripts/GetVariableList.lua
-- GetVariableList.lua, retrieves a list of the variables
local luaj = luajava -- just shorthand 
local script, ide = ... -- grab the arguments passed from Java via LuaJ

-- Get convenience function to generate file name
local Config = luaj.bindClass("org.supremica.properties.Config")
local getFileName = dofile(Config.FILE_SCRIPT_PATH:getValue():getPath().."/getFileName.lua")

-- local VariableComponentProxy = luaj.bindClass("net.sourceforge.waters.model.module.VariableComponentProxy")
-- if not VariableComponentProxy then print("VariableComponentProxy not fond") return end

local Helpers = luaj.bindClass("org.supremica.Lupremica.Helpers") 
if not Helpers then print("Lupremica.Helpers not found") return end

local module = ide:getActiveDocumentContainer():getEditorPanel():getModuleSubject()
if not module then print("No active module found") return end

local varlist = Helpers:getVariableList(module)
for i = 1, varlist:size() do
	local var = varlist:get(i-1)
	print(var:getIdentifier():toString().. " : "..
		var:getType():toString().." : "..
		var:getInitialStatePredicate():toString())
end

-- To retrieve also lists of the automata and events, uncomment this:
--[[
local automata = Helpers:getAutomatonList(module)
for i = 1, automata:size() do
	local aut = automata:get(i-1)
	print(aut:getName().." : "..aut:getKind():toString())
end

local eventDeclList = module:getEventDeclList()
for i = 1, eventDeclList:size() do
  local event = eventDeclList:get(i-1)
  print(event:getName())
end
--]]